### PR TITLE
Edit vector/rect properties in a "linked" manner when holding down Ctrl

### LINKED ISSUE #144: Add a link/unlink option for subproperties like "scale"
**Describe the project you are working on:**
2D action game, card game

**Describe the problem or limitation you are having in your project:**
Throughout the day I change properties like `scale` in the editor many, many times. Which already can be quite annoying because of https://github.com/godotengine/godot/issues/28618.
In 95% of the cases I want to change properties like `scale` proportionally. So I have to first set the x, then the y value, _both to the same value_. Because the editor does not offer to link those properties, all those operations have to be done twice ... every ... single ... time.

**Describe how this feature / enhancement will help you overcome this problem or limitation:**
A small clickable link/unlink icon next to values that would often be set proportionally. 
A small mouseover text explaining what it does, for those unsure about the meaning of the icon, similar to mouseover on properties.
I would expect `scale` to be **linked** by default, others may be unlinked by default.

**Show a mock up screenshots/video or a flow diagram explaining how your proposal will work:**
linked_properties

**Describe implementation detail for your proposal (in code), if possible:**
I'm C++ illiterate.

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
It's so common everywhere else, people are used to that level of comfort. I expect it to be used a lot.

**Is there a reason why this should be core and not an add-on in the asset library?:**
Noone will/would/should download addons for such small UI quality of life improvements.

### PR DESCRIPTION
**Pull request by Calinou**
**Don't merge yet.** This PR needs to be modified so it doesn't link coordinates when dragging properties to change their values while holding down <kbd>Ctrl</kbd>.

___

For Rect2, Rect2i and AABB, this only applies to the size, not the position.

This closes https://github.com/godotengine/godot-proposals/issues/144.

**Note:** When cherry-picking to the `3.2` branch, remove the parts that modify the integer vector types as these aren't present in 3.2.x.

### CODE REVIEW

File: editor/editor_properties.cpp [hunk 1]
```diff
@@ -1301,9 +1333,29 @@ void EditorPropertyVector3::_value_changed(double val, const String &p_name) {
 	}
 
 	Vector3 v3;
-	v3.x = spin[0]->get_value();
-	v3.y = spin[1]->get_value();
-	v3.z = spin[2]->get_value();
+	if (Input::get_singleton()->is_key_pressed(KEY_CONTROL)) {
+		// Set all components to the edited value to quickly set "linked" values such as scale.
+		if (p_name == "x") {
```
Comment (nathanfranke) [inline editor/editor_properties.cpp:1338]: `_value_changed` has a `val` parameter. Use that instead of whatever this is.

**Review by nathanfranke — CHANGES_REQUESTED** [review]

**Comment by nathanfranke** [comment]
I think that a link button would make more sense. I did a bit of work in my [fork](https://github.com/nathanfranke/godot/commit/d10fdeaa26a52a42f874fbafb45606c8c16075e3)

File: editor/editor_properties.cpp [hunk 2]
```diff
@@ -1301,9 +1333,29 @@ void EditorPropertyVector3::_value_changed(double val, const String &p_name) {
...
+			v3.x = spin[1]->get_value();
+			v3.y = spin[1]->get_value();
+			v3.z = spin[1]->get_value();
+		} else if (p_name == "z") {
+			v3.x = spin[2]->get_value();
+			v3.y = spin[2]->get_value();
+			v3.z = spin[2]->get_value();
+		} else {
+			ERR_PRINT("Unexpected component name (please report).");
+		}
```
Comment (nathanfranke) [inline editor/editor_properties.cpp:1352]:
```suggestion
		v3 = Vector3(val, val, val);
```

File: editor/editor_properties.cpp [hunk 3]
```diff
@@ -1564,9 +1646,29 @@ void EditorPropertyVector3i::_value_changed(double val, const String &p_name) {
...
```
Comment (nathanfranke) [inline editor/editor_properties.cpp:1665]: Same with above

**Review by nathanfranke — CHANGES_REQUESTED** [review]
Added the lines to change and suggestions (Thanks @aaronfranke)

**Comment by aaronfranke** [comment]
@Calinou What's the status of this PR? Do you plan to continue working on this? There are requested changes above that need to be addressed.

**Comment by akien-mga** [comment]
Superseded by #59125.

**Pull request closed without merging**